// @mikro-orm/knex/SqlEntityManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlEntityManager = void 0;
const core_1 = require("@mikro-orm/core");
const query_1 = require("./query");
/**
 * @inheritDoc
 */
class SqlEntityManager extends core_1.EntityManager {
    /**
     * Creates a QueryBuilder instance
     */
    createQueryBuilder(entityName, alias, type, loggerContext) {
        const context = this.getContext();
        return new query_1.QueryBuilder(entityName, this.getMetadata(), this.getDriver(), context.getTransactionContext(), alias, type, context, loggerContext ?? context.loggerContext);
    }
    /**
     * Shortcut for `createQueryBuilder()`
     */
    qb(entityName, alias, type, loggerContext) {
        return this.createQueryBuilder(entityName, alias, type, loggerContext);
    }
    /**
     * Returns configured knex instance.
     */
    getKnex(type) {
        return this.getConnection(type).getKnex();
    }
    async execute(queryOrKnex, params = [], method = 'all') {
        return this.getDriver().execute(queryOrKnex, params, method, this.getContext(false).getTransactionContext());
    }
    getRepository(entityName) {
        return super.getRepository(entityName);
    }
    applyDiscriminatorCondition(entityName, where) {
        // this is handled in QueryBuilder now for SQL drivers
        return where;
    }
}
exports.SqlEntityManager = SqlEntityManager;
